Feature#83_Create_tests_for_all_models

CheshirLvova opened the pull request:
Feature#144-tests_for_shipping_model

CheshirLvova committed:
Feature#83_Create_tests_for_all_models

Feature#144-tests_for_shipping_model

CheshirLvova committed:
Feature#92_Test_for_Order_model

strangerok1507 commented on ProjectShop/order/tests/tests.py:
```diff
@@ -0,0 +1,21 @@
+from django.test import TestCase
+from django.db.utils import IntegrityError
+
```
rename this file to test_models

CheshirLvova replied:
done)

strangerok1507 commented on ProjectShop/Shipping/tests/tests.py:
```diff
@@ -0,0 +1,19 @@
+from django.test import TestCase
+from django.db.utils import IntegrityError
+
```
rename this file to test_models

v-lavrushko commented on ProjectShop/order/tests/factories.py:
```diff
@@ -0,0 +1,30 @@
+    order = factory.SubFactory(OrderModelFactory)
+    #product = factory.SubFactory(ProductFactory)
+
+    order_items_qantity = factory.Faker('random_int', min=1, max=100)
```
*quantity

CheshirLvova replied:
fixed

v-lavrushko commented on ProjectShop/order/tests/tests.py:
```diff
@@ -0,0 +1,21 @@
+
+    def test_name(self):
+        OrderItems = OrderItemsModelFactory()
+        self.assertIsNotNone(OrderItems)
```
PEP 8 - blank line

MykhailoGrynenko commented on ProjectShop/Shipping/tests/tests.py:
```diff
@@ -0,0 +1,19 @@
+
+    def test_create(self):
+        shipping = ShippingFactory()
+        self.assertIsNotNone(shipping)
```
It seems like there is an indentation issue. What is the purpose of the setUp method, which does nothing?
![Screenshot from 2021-10-04 14-10-50](https://user-images.githubusercontent.com/56480204/135841630-d86df8cc-bd73-482b-986b-317100e4b030.png)

MykhailoGrynenko commented on ProjectShop/Shipping/tests/tests.py:
```diff
@@ -0,0 +1,19 @@
+
+
+
+
```
So many blank lines over here.

MykhailoGrynenko commented on ProjectShop/order/tests/factories.py:
```diff
@@ -0,0 +1,30 @@
+import datetime
+
+from order.models import OrderModel
+from order.models import OrderItemsModel
```
from order.models import OrderModel, OrderItemsModel

CheshirLvova replied:
done)

MykhailoGrynenko commented on ProjectShop/Shipping/tests/tests.py:
```diff
@@ -0,0 +1,19 @@
+from django.test import TestCase
+from django.db.utils import IntegrityError
+
+from Shipping.tests.factories import *
```
I believe importing an asterisk is not the best practice.

MykhailoGrynenko reviewed (changes requested):
Changes requested.

strangerok1507 commented on ProjectShop/Shipping/tests/factories.py:
```diff
@@ -0,0 +1,17 @@
+
+    user = factory.SubFactory(UserFactory)
+
+    postal_code = factory.Faker('45678')
```
что за провайдер 45678?

CheshirLvova committed:
Hotfix#83_Fixed_tests_for_all_models

Updated file names, added docstrings and some changes to models,, implemented new tests for order, order items and shipping models

CheshirLvova committed:
Hotfix#83_Fixed_tests_for_all_models_upd

Updated assertLess functions because of invalid input parameters

CheshirLvova committed:
Hotfix#83_Fixed_tests_for_all_models_upd_fixed_order_model

Updated order model due the merge and added new tests to order model, fixed tabs in factory to shipping model part

CheshirLvova committed:
fixed some tests for order and shipping models

CheshirLvova committed:
Merge branch 'main' into feature#83-create-tests-for-all-models

Maxgioman commented on ProjectShop/Shipping/tests/test_models.py:
```diff
@@ -0,0 +1,35 @@
+
+    def test_country_str(self):
+        shipping = self.shipping
+        self.assertEqual(shipping.country, shipping.__str__())
```
як у нас один і той же метод __str__() видає три різні варіанти в 3 тестах і всі правильні?

MykhailoGrynenko committed:
Merge pull request #221 from kolyasalubov/dev

Third Demo Release.

CheshirLvova committed:
Merge branch 'main' into feature#83-create-tests-for-all-models

CheshirLvova committed:
Updated files due new models titles
